### PR TITLE
Doc: add change history to documentation (Infra)

### PR DESCRIPTION
**Pull request by tang-mm**
## Description

Add a change history page in the Checkbox reference docs, containing a link to the GitHub Releases page. 

This initial version of change history is a copy of the GitHub-generated release notes. With the implementation of PR title validation, further changes to this page should be improved into an integrated action within the release process.

## Resolved issues

[CHECKBOX-846](https://warthogs.atlassian.net/browse/CHECKBOX-846)

## Tests

1. In the `docs/` folder, build the documentation set locally by running:
```
$ make install
$ make run
```
2. Go to <http://127.0.0.1:8000/reference/changes> 

[CHECKBOX-846]: https://warthogs.atlassian.net/browse/CHECKBOX-846?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by yphus — APPROVED** [review]
tested locally, the changelog renders very well, thx.

**Comment by tang-mm** [comment]
appended a commit to fix spelling check errors

File: docs/.sphinx/wordslist.txt [hunk 1]
```diff
@@ -51,32 +58,52 @@ libre
...
 ng
+Nvidia
+OEM
+OSI
+OTG
 parametrize
 Parsers
 pdf
 Permalink
+Permalink
```
Comment (kissiel) [inline docs/.sphinx/wordslist.txt:81]: The only diff with the previous line is the lack of space at the end.

**Review by kissiel — COMMENTED** [review]
Some of the lines you've added don't end with a blank space, and from what I can tell, the existing ones do. I pointed out one easy to spot below.
I'd like to know if there's a reason to have them, if not, I'll gladly trim all of them.

**Comment by tang-mm** [comment]
Following the discussion in [MM](https://chat.canonical.com/canonical/pl/mb54zso4xtd3xgp8scfiu5ei5y), the initial changelog only contains a link for consistency and maintenance purposes. Further content should be generated through automation in the release process.

(removed `docs/.wordlist.txt`  because it is replaced by `docs/sphinx/wordlist.txt`)

**Review by Hook25 — APPROVED** [review]
Ty for the contribution and removing that old wordlist as well

**Pull request merged**